torch/csrc/jit/ir/ir.cpp

```diff
@@ -24,23 +24,38 @@
 namespace torch {
 namespace jit {
 
+namespace {
+
 // Constants relating to maintaining the topological index of nodes.
 //
 // Lower and upper bounds of the index. Inclusive range.
-static constexpr topo_position_t kLowerBound = INT64_MIN;
-static constexpr topo_position_t kUpperBound = INT64_MAX;
-static constexpr topo_position_t kMidPoint = 0;
+constexpr topo_position_t kLowerBound = INT64_MIN;
+constexpr topo_position_t kUpperBound = INT64_MAX;
+constexpr topo_position_t kMidPoint = 0;
 
 // How far away to space nodes that are appended to the graph.
 // should be 2^n, where:
 //   - n is the maximum number of repeated insertions without a re-index
 //   - 2^(64-n) is the maximum number of appends to the end without reindex
-static constexpr topo_position_t kAppendInterval = 1099511627776ULL /* 2^40 */;
+constexpr topo_position_t kAppendInterval = 1099511627776ULL /* 2^40 */;
 
-static void printValueRef(std::ostream& out, const Value* n) {
+void printValueRef(std::ostream& out, const Value* n) {
   out << "%" << n->debugName();
 }
 
+bool isNumber(c10::string_view str) {
+  return str.find_first_not_of("0123456789") == std::string::npos;
+}
+
+std::string normalizeAttrName(c10::string_view field) {
+  if (isNumber(field)) {
```

Comment:
Attribute names that are numbers are usually illegal in Python unless they are manually inserted into the object's instance dictionary. Why do you need to check whether `field` is a number?

Reply:
I see that there are some existing tests that manage to use attributes starting with numbers, but I am wondering how...

Reply:
@SplitInfinity 
I'm seeing "0" used as attr name in ads unit tests... https://www.internalfb.com/intern/testinfra/diagnostics/5910974564491719.281474990955546.1622163900/

After some investigation, I found there's indeed one way:
```
class M(nn.Module):
     def __init__(self):
           setattr(self, "0", 0)
     def forward(self):
           return getattr(self, "0")
```
this way we got something like:
```
graph(%self : __torch__.M):
  %_0 : int = prim::GetAttr[name="0"](%self)
  return (%9)
```
and as the test result shows, this happens in real model...

Reply:
Lol